### PR TITLE
Truncate long course and lesson titles

### PR DESCRIPTION
**Pull request by katevoss**
Closes https://github.com/catalpainternational/asteroid/issues/189

### CODE REVIEW

File: src/riot/WagtailPages/CoursePage.riot.html [hunk 1]
```diff
@@ -104,8 +96,10 @@ <h5 class="section-title">{ TRANSLATIONS.course_exam() }</h5>
             stickyHeader() {
                 if (window.pageYOffset > this.state.header.offsetTop) {
                     this.state.header.classList.add("sticky");
+                    this.state.header.children[0].children[2].classList.add("truncate-if-long");
```
Comment (sandroroux) [inline src/riot/WagtailPages/CoursePage.riot.html:99]: This level of nested access gives me the heebie-jeebies. Can you give the element an`id` or `class` and query it directly? Imagine if we ever restructure the HTML.

File: src/riot/WagtailPages/CoursePage.riot.html [hunk 2]
```diff
@@ -104,8 +96,10 @@ <h5 class="section-title">{ TRANSLATIONS.course_exam() }</h5>
...
                 } else {
                     this.state.header.classList.remove("sticky");
+                    this.state.header.children[0].children[2].classList.remove("truncate-if-long");
```
Comment (sandroroux) [inline src/riot/WagtailPages/CoursePage.riot.html:102]: Same comment as above.

File: src/riot/WagtailPages/CoursePage.riot.html (hunk 1 above)
Comment (katevoss) [reply src/riot/WagtailPages/CoursePage.riot.html]: Changes made ✅

File: src/riot/WagtailPages/CoursePage.riot.html [hunk 3]
```diff
@@ -9,24 +9,16 @@
...
-                duration="{lesson.duration}"
-                link="{lesson.id}"
-                isCompleted="{isLessonComplete(lesson)}"
-                isComingSoon="{lesson.coming_soon}"
-                showFooter="{true}"
-                type="lesson"
-            ></Card>
+            <Card each="{lesson in props.wagtailPage.lessons}" title="{lesson.title}" description="{lesson.description}"
+                duration="{lesson.duration}" link="{lesson.id}" isCompleted="{isLessonComplete(lesson)}"
+                isComingSoon="{lesson.coming_soon}" showFooter="{true}" type="lesson"></Card>
```
Comment (sandroroux) [inline src/riot/WagtailPages/CoursePage.riot.html:21]: This should be `prettier`ed.

File: src/riot/WagtailPages/LessonPage.riot.html [hunk 4]
```diff
@@ -23,53 +23,27 @@ <h2>{props.wagtailPage.title}</h2>
...
-                <a
-                    if="{props.wagtailPage.data.long_description && !state.showLongDescription}"
-                    onclick="{toggleLessonDescription}"
-                    translate
-                    >See more</a
-                >
+                <a if="{props.wagtailPage.data.long_description && state.showLongDescription}"
+                    onclick="{toggleLessonDescription}" translate>See less</a>
+                <a if="{props.wagtailPage.data.long_description && !state.showLongDescription}"
+                    onclick="{toggleLessonDescription}" translate>See more</a>
```
Comment (sandroroux) [inline src/riot/WagtailPages/LessonPage.riot.html:29]: `prettier`

File: src/riot/WagtailPages/CoursePage.riot.html [hunk 5]
```diff
@@ -9,24 +9,16 @@
                     <div>
                         <p>{ TRANSLATIONS.course() }</p>
                     </div>
-                    <h3>{props.wagtailPage.data.title}</h3>
+                    <h3 id="title">{props.wagtailPage.data.title}</h3>
```
Comment (PeteCoward) [inline src/riot/WagtailPages/CoursePage.riot.html:12]: I'd rather this id was  more specific. It's really easy to accidentally end up with multiple 'title' ids in the DOM when using such generic ids.
courseTitle, or even coursePageTitle might be better, same for the lesson page below

Comment (PeteCoward) [reply src/riot/WagtailPages/CoursePage.riot.html:12]: or course-title or coursepage-title

**Pull request merged**